cachi2/core/package_managers/yarn/project.py

```diff
@@ -31,43 +39,56 @@ def cache_path(self) -> str:
 
         Fallback to the default path in case the configuration key is missing.
         """
-        return NotImplemented
+        return self._data.get("cacheFolder", "./.yarn/cache")
 
     @property
     def registry_server(self) -> str:
         """Get the globally configured registry server.
 
         Fallback to the default server in case the configuration key is missing.
         """
-        return NotImplemented
+        return self._data.get("npmRegistryServer", "https://registry.yarnpkg.com")
 
     @property
     def yarn_path(self) -> Optional[str]:
         """Path to the yarn script present in this directory."""
-        return NotImplemented
+        try:
+            return self._data.get("yarnPath")
+        except KeyError:
+            log.warning(f"No yarnPath in {self._path}")
```

Comment:
Since yarnPath is not a mandatory key, I think logging a warning here may be misleading.

Reply:
+1

also `self._data.get()` will never raise a KeyError, IMO you can remove the whole try/except

Reply:
removed